src/main/scala/fm/xml/XmlReaderWriter.scala

```diff
@@ -17,29 +17,55 @@ package fm.xml
 
 import fm.common.{FileUtil, InputStreamResource, Resource, SingleUseResource}
 import java.io.{File, InputStream, OutputStream, Reader, StringReader}
-import scala.reflect.{ClassTag, classTag}
+import scala.reflect.ClassTag
 
-final case class XmlReaderWriter[T: ClassTag](rootName: String, itemName: String, defaultNamespaceURI: String = "", overrideDefaultNamespaceURI: String = "") {
-  private[this] val classes: Seq[Class[_]] = Seq(classTag[T].runtimeClass)
-  
+object XmlReaderWriter {
+  // Apply helper, e.g. XmlReaderWriter("root")(Path1, Path2).reader(is)
+
+  def apply[T](
+    rootName: String,
+    defaultNamespaceURI: String = "",
+    overrideDefaultNamespaceURI: String = ""
+  )(target: XmlReaderPath[_, T], rest: XmlReaderPath[_, T]*): XmlReaderWriter[T] = {
+    XmlReaderWriter(rootName, defaultNamespaceURI, overrideDefaultNamespaceURI, (target +: rest):_*)
+  }
+
+  // Semi-legacy-compatible XmlReaderWriter for a single itemName,
+  // the original had default argument values for defaultNamespaceURI AND overrideDefaultNamespaceURI
+  // that interferes with the new default argugment constructor
+  def apply[T: ClassTag](rootName: String, itemName: String): XmlReaderWriter[T] = apply(rootName, itemName, defaultNamespaceURI = "", overrideDefaultNamespaceURI = "")
+
+  def apply[T: ClassTag](
+    rootName: String,
+    itemName: String,
+    defaultNamespaceURI: String,
+    overrideDefaultNamespaceURI: String
+  ): XmlReaderWriter[T] = {
+    new XmlReaderWriter(rootName, defaultNamespaceURI, overrideDefaultNamespaceURI, XmlReaderPath(itemName))
+  }
+}
+
+final case class XmlReaderWriter[T](rootName: String, defaultNamespaceURI: String, overrideDefaultNamespaceURI: String, targets: XmlReaderPath[_, T]*) {
```

Comment:
As far as I can tell the `XmlReaderPath` is NOT compatible with the `XmlWriter`.